src/ggrc/snapshotter/helpers.py

```
@@ -0,0 +1,275 @@
+                revision_id_cache[key] = revid
+    return revision_id_cache
+
+
+def get_relationships(relationships):
+  """Retrieve relationships
+
+  Args:
+    relationships:
+  """
+  with benchmark("snapshotter.helpers.get_relationships"):
+    if relationships:
+      relationship_columns = db.session.query(
+          models.Relationship.id,
+          models.Relationship.modified_by_id,
+          models.Relationship.created_at,
+          models.Relationship.updated_at,
+          models.Relationship.source_type,
+          models.Relationship.source_id,
+          models.Relationship.destination_type,
+          models.Relationship.destination_id,
+          models.Relationship.context_id,
+      )
+
+      return relationship_columns.filter(
+          tuple_(
+              models.Relationship.source_type,
+              models.Relationship.source_id,
+              models.Relationship.destination_type,
+              models.Relationship.destination_id,
+          ).in_(relationships)
+      ).union(
+          relationship_columns.filter(
+              tuple_(
+                  models.Relationship.destination_type,
+                  models.Relationship.destination_id,
+                  models.Relationship.source_type,
+                  models.Relationship.source_id
+              ).in_(relationships)
+          )
+      )
+    else:
+      return set()
+
+
+def get_event(_object, action):
+  """Retrieve the last event for parent objects that performed
+  PUT/POST/IMPORT/Event.action action."""
+  with benchmark("snapshotter.helpers.get_event"):
+    event = db.session.query(
+        models.Event.id,
+        models.Event.resource_type,
+        models.Event.resource_id).filter(
+        models.Event.resource_type == _object.type,
+        models.Event.resource_id == _object.id,
+        models.Event.action == action).order_by(
+        models.Event.id.desc()).first()
+    if not event:
+      raise Exception("No event found!")
```

What if multiple events are found and you're only returning the first one. I think this is a bad code smell.

If there shoud be one and only one event, and if we need to raise an exception if an event is not found. you should use `.one()` instead of `.first()` because the `.one()` will raise an exception if the query does not return exactly one result.

This way you can avoid the whole try except block